android/KMEA/app/src/main/java/com/tavultesoft/kmea/KeyboardPickerActivity.java

```diff
@@ -511,11 +519,13 @@ protected static void deleteKeyboard(Context context, int position) {
       }
       if (position == curKbPos && listView != null) {
         switchKeyboard(0);
-      } else {
+      } else if(listView != null) { // A bit of a hack, since LanguageSettingsActivity calls this method too.
         curKbPos = getCurrentKeyboardIndex();
         setSelection(curKbPos);
       }
     }
+
+    notifyKeyboardsUpdate(context);
```

Comment:
In case it's not otherwise clear here, most of this file's changes kind of "Frankenstein" the old picker's management with the new adapters, rather than adding extra refactoring work here.  The PR and its chain are large enough already.